matchbox_socket/src/webrtc_socket/socket.rs

```diff
@@ -100,16 +100,26 @@ pub struct WebRtcSocketConfig {
     pub attempts: Option<u16>,
 }
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+/// The state of a connection to a peer
+pub enum PeerState {
+    // todo: add connecting state?
```

Comment:
New, Connecting, Connected, Disconnected, Closed, Error/Unknown would be all the states we should care about. (New would not trigger any state changes, though)

Reply:
Currently, we can't get to them on wasm anyway, so this is now mostly about having another source of detecting "failure", see comments I added in latest version